torch/_refs/__init__.py

```diff
@@ -2728,6 +2730,61 @@ def _normalize(
     return out, mean, rstd
 
 
+@register_decomposition(torch.ops.aten.native_group_norm.default, disable_meta=True)
+def native_group_norm(
+    input: Tensor,
+    weight: Optional[Tensor],
+    bias: Optional[Tensor],
+    batch_size: int,
+    num_channels: int,
+    flattened_inner_size: int,
+    num_groups: int,
+    eps: float,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    utils.check(
+        input.ndim >= 2,
+        lambda: f"Expected at least 2 dimensions for input tensor but recieved {input.ndim}",
+    )
+    utils.check(
+        num_channels % num_groups == 0,
+        lambda: "Expected number of channels in input to be divisible by num_groups, "
+        + f"but got input of shape {input.shape} and num_groups = {num_groups}",
+    )
+
+    reduction_dims = [-2, -1]
+    input_reshaped = torch.reshape(
+        input,
+        [batch_size, num_groups, num_channels // num_groups, flattened_inner_size],
+    )
+    out, mean, rstd = _normalize(input_reshaped, reduction_dims, eps)
+    out = out.view(input.shape)
+
+    broadcast_dims = [0] + list(dim for dim in range(2, input.ndim))
+    unsqueeze_bias = None
+    if bias is not None:
+        unsqueeze_bias = prims.expand_dims(bias, broadcast_dims, input.ndim)
+    unsqueeze_weight = None
+    if weight is not None:
+        unsqueeze_weight = prims.expand_dims(weight, broadcast_dims, input.ndim)
+
+    if unsqueeze_weight is not None:
+        out = out * unsqueeze_weight
+    if unsqueeze_bias is not None:
+        out = out + unsqueeze_bias
+
+    if out.dtype != input.dtype:
```

Comment:
Use `maybe_convert_to_dtype`, which performs these checks for you

https://github.com/pytorch/pytorch/blob/4b757f4633494d7bbc55973f36f14aeca96387fa/torch/_prims_common/wrappers.py#L20

Reply:
`mypy` fails because it expects `TensorLikeType` but `_maybe_convert_to_dtype` returns `Union[TensorLikeType, NumberType, Sequence, None]`. I'll add a comment for this.

Reply:
Just direct mypy to ignore the line